AzureCalling/External/Calling/CallingContext.swift

```diff
@@ -334,6 +338,13 @@ class CallingContext: NSObject {
     private func setupRemoteParticipantsEventsAdapter() {
         participantsEventsAdapter = ParticipantsEventsAdapter()
 
+        participantsEventsAdapter?.onIsMutedChanged = { [weak self] _ in
+            guard let self = self else {
+                return
+            }
+            self.notifyRemoteParticipantsUpdated()
```

Comment:
1. Can we do a filter here to see if the participant with muting state change in displayed in the calling view grid.
 If he is not in the grid, we don't need to notify the views updates.  
2. Would it be kind of waste of resource that every time a small muting icon change on one participant view triggers all participant view updating? Is it possible to implement business logic to find the participant view need updates and only update that specific view in the Calling View controllers?

Reply:
Added another event to target only active participant that is on screen if triggered and update the participant's mute indicator